src/sqlfluff/core/dialects/base.py

```diff
@@ -153,6 +153,13 @@ def replace(self, **kwargs):
                 )
             self._library[n] = kwargs[n]
 
+    def get(self, name):
```

Comment:
Lol - Turns out this was a much harder problem to address. The issue was that this could return almost anything and validating that was a PITA.

I've split this into two methods - one for grammars and one for segments so that you can at least infer what you're getting.